custom_components/tuya_local/devices/eeese_anna_dehumidifier.yaml

```diff
@@ -0,0 +1,170 @@
+name: Eeese Anna dehumidifier
+primary_entity:
+  entity: humidifier
+  class: dehumidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: "mdi:air-humidifier-off"
+          icon_priority: 2
+        - dps_val: true
+          icon: "mdi:air-humidifier"
+          icon_priority: 4
+    - id: 2
+      name: humidity
+      type: integer
+      range:
+        min: 30
+        max: 80
+      mapping:
+        - step: 5
+    - id: 5
+      name: mode
+      type: string
+      mapping:
+        - dps_val: Manual
+          value: normal
+        - dps_val: Auto
+          value: auto
+        - dps_val: Laundrydrying
+          value: laundry
+          icon: "mdi:tshirt-crew"
+          icon_priority: 3
+        - dps_val: Sleeping
+          value: sleep
+          icon: "mdi:sleep"
+          icon_priority: 3
+        - dps_val: Purifying
+          value: purify
+          icon: "mdi:air-purifier"
+          icon_priority: 3
+    - id: 6
+      type: integer
+      name: current_humidity
+    - id: 19
+      type: bitfield
+      name: error
+      optional: true
+      mapping:
+        - dps_val: 4
+          icon: "mdi:cup-water"
+          icon_priority: 1
+secondary_entities:
+  - entity: sensor
+    name: Current humidity
+    deprecated: humidifier
+    category: diagnostic
+    class: humidity
+    dps:
+      - id: 6
+        type: integer
+        name: sensor
+        unit: "%"
+        class: measurement
+  - entity: binary_sensor
+    name: Tank
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 19
+        type: bitfield
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: 0
+            value: false
+          - dps_val: null
+            value: false
+          - value: true
```

Comment:
Tank should be one fault code (4 based on the above error attribute, but if that was just copied from another config, confirm it first).
I would split this into two binary_sensors, one for the routine Tank full indication (true for dps_val: 4, otherwise false), and one for actual faults (false for 0, 4, null, true for other values)

Reply:
I split it into Fault, Tank, Defrost, in accordance with the bitmap i got from the Tuya's Developer Portal.
`{
  "label": [
    "E1",
    "E2",
    "tankfull",
    "defrost"
  ]
}`, for dps_value 1+2, 4, 8.